Move code out of main.py

0ddlyoko opened the pull request:
This PR mainly moves the code that was in main.py in correct place.
- Code that was in main.py is now in correct place
- Add entity/Entity.py and basic classes for entity:
    - entity id
    - entity location
    - ...
-  Add EntityManager class to load and instanciate an Entity
- Add mcPy/McPy.py which is the "launcher" file, it retrieves some data and launch the server (or launch tests)
- Add mcPy/MultiProcessing.py:
    - This class will handle the processing stuff of McPy
- Add mcPy/Parser.py to parse the arguments given when starting the server
- Add mcPy/Server.py:
    - This class is the main class, it will initialize all stuff to launch the server.
    - (It's like the "motherboard" of a computer)
- Rewrite the network stuff. Now we can send and receive packets
- Add player/Player.py:
    - This class contains basic information about a player and player stuff
- Add utils/Scheduler.py:
    - Contains basic stuff to schedule some action (like executing an action after x ticks)
- Add utils/Thread.py:
    - Contains AtomicInteger, a class used to have an atomic auto increment counter
- Add utils/Vector.py:
    - Contains basic stuff for 2d and 3d vector

<3

0ddlyoko committed:
[IMP] Refractor the code

This PR mainly moves the code that was in main.py in correct place.
- Code that was in main.py is now in correct place
- Add entity/Entity.py and basic classes for entity:
    - entity id
    - entity location
    - ...
- Add EntityManager class to load and instanciate an Entity
- Add mcPy/McPy.py which is the "launcher" file, it retrieves some data and launch the server (or launch tests)
- Add mcPy/MultiProcessing.py:
    - This class will handle the processing stuff of McPy
- Add mcPy/Parser.py to parse the arguments given when starting the server
- Add mcPy/Server.py:
    - This class is the main class, it will initialize all stuff to launch the server.
    - (It's like the "motherboard" of a computer)
- Rewrite the network stuff. Now we can send and receive packets
- Add player/Player.py:
    - This class contains basic information about a player and player stuff
- Add utils/Scheduler.py:
    - Contains basic stuff to schedule some action (like executing an action after x ticks)
- Add utils/Thread.py:
    - Contains AtomicInteger, a class used to have an atomic auto increment counter
- Add utils/Vector.py:
    - Contains basic stuff for 2d and 3d vector

<3

0ddlyoko committed:
[IMP] Move Server.py to avoid circular import

Moves the Server.py file to avoid having circular import and add typing.

Moving Server.py inside classes directory allows the use of typing for this class and avoid circular imports.

0ddlyoko committed:
[IMP] Add typing for Server class

This commit add imports for typing for the `Server` class

Geolykt commented on .gitignore:
```diff
@@ -2,7 +2,7 @@
 .vscode
 __pycache__
-McPy/
+/McPy/
```
Not entirely sure how gitignores work, but are you sure this isn't based on the root directory if a slash is preceding?

0ddlyoko replied:
/McPy/ is based on root directory but not McPy/

Geolykt replied:
Ah, so then it's wanted, but why though?

0ddlyoko replied:
When running startup.py file, it downloads the project and put it in a directory called McPy, this was probably why `McPy/` was in .gitignore
Here, because this line exists, it was ignoring directory at `/classes/McPy`.
I can remove the line if you want, for now if we don't launch the file the directory isn't created

0ddlyoko replied:
Done, I've removed this line

Geolykt commented on classes/mcPy/McPy.py:
```diff
@@ -0,0 +1,63 @@
+
+def _launch(parser: Parser):
+    try:
+        logging.info("The Blackfire probe")
```
This makes little to no sense as it's repeated again if it's not installed (makes a bit more sense if it is installed though)

0ddlyoko replied:
Nice catch :D

0ddlyoko replied:
Done

Geolykt commented on classes/mcPy/MultiProcessing.py:
```diff
@@ -0,0 +1,79 @@
+
+    def start(self):
+        self.started = True
+        for p in self.workers:
+            logging.info('Starting worker')
+            p.start()
+            logging.info('Worker started')
```
maybe numerate the workers or something, it can be a bit confusing to see workers start up multiple times.

0ddlyoko replied:
Done

Geolykt commented on classes/mcPy/McPy.py:
```diff
@@ -0,0 +1,63 @@
+    server.start()
+
+    # End Network stuff
+    logging.info("Shutting down server!")
```
server.stop() is already printing a similar message, either remove it from there, or vice-versa

0ddlyoko replied:
Done

Geolykt commented on classes/utils/Vector.py:
```diff
@@ -0,0 +1,90 @@
+from math import sqrt, floor
+
+
+class Vector2:
```
I'd use Vector2D here

0ddlyoko replied:
Done, I've also changed Vector3 to Vector3D

Geolykt reviewed (changes requested):
With the code I only have a few issues, however there are more issues that I cannot really give the source.
Additional issues:
  - Pinging is broken both on 1.15.2 and 1.16.4
  - Chat no longer seems to be transmitted to the console (although idk if that change was introduced here)

0ddlyoko committed:
[IMP] Reviews fix

- Remove useless line in .gitignore
- Remove 2 repeated log message
- Add name to processes & threads
- Change Vector2 and Vector3 to Vector2D and Vector3D

0ddlyoko committed:
[FIX] Display chat on console

When a player was writing, nothing appears on the console.
This commit fix this bug by printing the message on the console

Fix also a typo

0ddlyoko commented:
Chat is now displayed on the console but should be refactored in the core when someone will start working with chat.
Ping still displays an error but this error exists before this refactoring

Geolykt reviewed (approved):
Working as intended

0ddlyoko commented:
Merged in develop !
